src/phy/fuzz_injector.rs

```diff
@@ -0,0 +1,120 @@
+use Result;
+use phy::{self, DeviceCapabilities, Device};
+use time::Instant;
+
+// This could be fixed once associated consts are stable.
+const MTU: usize = 1536;
+
+/// Represents a fuzzer. It is expected to either generate or contain data, which it replaces bytes
+/// in the packet with.
+pub trait Fuzzer {
+    /// Modify a single packet.
+    fn fuzz_packet(&self, packet_data: &mut [u8]);
+}
+
+/// A fuzz injector device.
+///
+/// A fuzz injector is a device that alters packets traversing through it according to the
+/// directions of a guided fuzzer. It is designed to support fuzzing internal state machines inside
+/// smoltcp, and is not for production use.
+#[allow(unused)]
+#[derive(Debug)]
+pub struct FuzzInjector<D: for<'a> Device<'a>, FTx: Fuzzer, FRx: Fuzzer> {
+    inner:   D,
+    fuzz_tx: FTx,
+    fuzz_rx: FRx,
+}
+
+#[allow(unused)]
+impl<D: for<'a> Device<'a>, FTx: Fuzzer, FRx: Fuzzer> FuzzInjector<D, FTx, FRx> {
+    /// Create a fuzz injector device, which defaults to passing the packets unaltered.
```

Comment:
This doesn't look right--it doesn't default to anything since it's not configurable...

Reply:
I was referring to the behavior of the Fuzz object. I will clarify this.